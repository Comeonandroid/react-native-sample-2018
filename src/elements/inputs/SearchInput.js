import React, { PureComponent } from 'react'
import { Keyboard } from 'react-native'
import PropTypes from 'prop-types'
import styled, { withTheme } from 'styled-components'
import { prop } from 'styled-tools'

import CustomIcon from '../../utils/icon'

import { View } from '../index'
import BaseInput from './BaseInput'

const inputStyles = theme => ({
  fontFamily: theme.typo.primaryFont,
  fontStyle: 'italic',
  color: theme.colors.inputSearchPlaceholder,
  fontSize: 12,
  marginLeft: 10,
  marginRight: 10
})

class SearchInput extends PureComponent {

  constructor(props) {
    super(props)
    this.handleCancelPress = this.handleCancelPress.bind(this)
  }

  handleCancelPress() {
    Keyboard.dismiss()
    this.props.onChange(this.props.name, '')
  }

  render() {
    const { theme, value, ...props } = this.props
    return (
      <Wrapper>
        <SearchIcon
          name={'search'}
          size={14}
        />
        <Input
          placeholderTextColor={theme.colors.inputSearchPlaceholder}
          underlineSize={0}
          textInputStyle={inputStyles(theme)}
          value={value}
          selectionColor={theme.colors.success}
          {...props}
        />
        {value.length > 0 &&
          <CancelIcon
            name={'cancel'}
            size={14}
            onPress={this.handleCancelPress}
          />
        }
      </Wrapper>
    )
  }
}

SearchInput.propTypes = {
  children: PropTypes.node,
  theme: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  name: PropTypes.string
}

const Wrapper = styled(View) `
  display: flex;
  flex-direction: row;
  padding: 12px 10px;
  align-items: center;
  justify-content: center;
  background: ${prop('theme.colors.inputSearchBackground')}
`

const SearchIcon = styled(CustomIcon) `
  color: ${prop('theme.colors.success')}
`

const CancelIcon = styled(CustomIcon) `
  color: ${prop('theme.colors.danger')}
`

const Input = styled(BaseInput) `
  flex: 1;
`

export default withTheme(SearchInput)
