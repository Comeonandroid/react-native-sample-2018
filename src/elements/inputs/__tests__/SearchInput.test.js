import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../specs/shared_examples/index'

import SearchInput from '../SearchInput'

const setup = propOverrides => {
  const props = Object.assign({
    name: 'name',
    value: '',
    children: null,
    onChange: jest.fn()
  }, propOverrides)

  const component = <SearchInput {...props} />

  return {
    props,
    component
  }
}

describe('<SearchInput />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  describe('shows close icon if val non null', () => {
    const { component } = setup({ value: 'Test' })
    itBehavesLike('aThemedComponent', { subject: component })
  })
})
