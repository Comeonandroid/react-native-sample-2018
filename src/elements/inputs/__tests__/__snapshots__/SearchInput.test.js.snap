// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#323131",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 12,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 12,
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 14,
        },
        Array [
          Object {
            "color": "#fcfcfc",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChange={[MockFunction]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#7b7c7d"
      secureTextEntry={undefined}
      selectionColor="#2a986f"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "marginTop": 0,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          Object {
            "color": "rgb(117, 117, 117)",
            "fontSize": 16,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          Object {
            "color": "#7b7c7d",
            "fontFamily": "Open Sans",
            "fontSize": 12,
            "fontStyle": "italic",
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(63,81,181, 0.9)",
            "height": 0,
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<SearchInput /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#5d5959",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 12,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 12,
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 14,
        },
        Array [
          Object {
            "color": "#4b4c4d",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChange={[MockFunction]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#fff"
      secureTextEntry={undefined}
      selectionColor="#2a986f"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "marginTop": 0,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          Object {
            "color": "rgb(117, 117, 117)",
            "fontSize": 16,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          Object {
            "color": "#fff",
            "fontFamily": "Open Sans",
            "fontSize": 12,
            "fontStyle": "italic",
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(63,81,181, 0.9)",
            "height": 0,
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<SearchInput /> shows close icon if val non null dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#323131",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 12,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 12,
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 14,
        },
        Array [
          Object {
            "color": "#fcfcfc",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChange={[MockFunction]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#7b7c7d"
      secureTextEntry={undefined}
      selectionColor="#2a986f"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "marginTop": 0,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          Object {
            "color": "rgb(117, 117, 117)",
            "fontSize": 16,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          Object {
            "color": "#7b7c7d",
            "fontFamily": "Open Sans",
            "fontSize": 12,
            "fontStyle": "italic",
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="Test"
    />
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(63,81,181, 0.9)",
            "height": 0,
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 14,
        },
        Array [
          Object {
            "color": "#fcfcfc",
          },
          Array [
            Object {
              "color": "#f25c74",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`<SearchInput /> shows close icon if val non null light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#5d5959",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 12,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 12,
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 14,
        },
        Array [
          Object {
            "color": "#4b4c4d",
          },
          Array [
            Object {
              "color": "#2a986f",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChange={[MockFunction]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#fff"
      secureTextEntry={undefined}
      selectionColor="#2a986f"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "marginTop": 0,
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          Object {
            "color": "rgb(117, 117, 117)",
            "fontSize": 16,
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          Object {
            "color": "#fff",
            "fontFamily": "Open Sans",
            "fontSize": 12,
            "fontStyle": "italic",
            "marginLeft": 10,
            "marginRight": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="Test"
    />
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(63,81,181, 0.9)",
            "height": 0,
            "left": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 14,
        },
        Array [
          Object {
            "color": "#4b4c4d",
          },
          Array [
            Object {
              "color": "#f25c74",
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
